src/libsystemd-network/dhcp-identifier.c

```diff
@@ -207,48 +203,20 @@ int dhcp_identifier_set_duid(
 }
 
 int dhcp_identifier_set_iaid(
-                int ifindex,
+                sd_device *dev,
                 const struct hw_addr_data *hw_addr,
                 bool legacy_unstable_byteorder,
-                bool use_mac,
                 void *ret) {
 
-        /* name is a pointer to memory in the sd_device struct, so must
-         * have the same scope */
-        _cleanup_(sd_device_unrefp) sd_device *device = NULL;
         const char *name = NULL;
         uint32_t id32;
         uint64_t id;
-        int r;
 
-        assert(ifindex > 0);
         assert(hw_addr);
         assert(ret);
 
-        if (udev_available() && !use_mac) {
```

Comment:
Previously this was conditionalized on `use_mac`, which was (usually) set to `client->test_mode`. Now this is unconditional if `dev` is set. Can you comment a bit on the effect?